### ISSUE TITLE
cluster-monitoring-operator stuck in CreateContainerConfigError state after upgrading from 4.5 to 4.6

### ISSUE THREAD

**Issue by kcchu** [openshift/okd#434]
**Describe the bug**
I upgraded a cluster from 4.5 to 4.6 via web console. The upgrade completed (all operators became Available in `oc get clusteroperators`). However, the `cluster-monitoring-operator` is found stuck in `CreateContainerConfigError` state. Killing the pods didn't solve the issue. Also, the other pods in `openshift-monitoring` namespace (such as prometheus) keep restarting even they are apparently healthy. 

As a side notes, the cluster is consuming a lot more CPU after upgrading. Each master is eating 1 of 4 vCPU allocated to it. I don't know what is eating the CPU yet, but I am guessing pods in `openshift-monitoring` contributed a lot.

Describe cluster-monitoring-operator pods shows these events:

```
Events:                   
  Type     Reason          Age                   From               Message                                  
  ----     ------          ----                  ----               -------                                  
  Normal   Scheduled       92m                   default-scheduler  Successfully assigned openshift-monitoring/cluster-monitoring-operator-f866d58f7-85hf6 to master2                                                      
  Normal   AddedInterface  92m                   multus             Add eth0 [10.130.0.33/23]                
  Normal   Pulled          92m                   kubelet            Container image "quay.io/openshift/okd-content@sha256:59fb32e264a6bb2f5a08a1280e7f1edbef76fd159ba03d61282d1766d43037d2" already present on machine     
  Normal   Created         92m                   kubelet            Created container cluster-monitoring-operator                                                                                                          
  Normal   Started         92m                   kubelet            Started container cluster-monitoring-operator                                                                                                          
  Warning  Failed          90m (x11 over 92m)    kubelet            Error: container has runAsNonRoot and image has non-numeric user (nobody), cannot verify user is non-root                                              
  Normal   Pulled          116s (x417 over 92m)  kubelet            Container image "quay.io/openshift/okd-content@sha256:1aa5bb03d0485ec2db2c7871a1eeaef83e9eabf7e9f1bc2c841cf1a759817c99" already present on machine  
```

**Version**
4.6.0-0.okd-2020-12-12-135354

**How reproducible**
My only test cluster. Didn't attempt to reproduce

**Log bundle**
https://encl.io/t/AZccK66jYDU#bnls-J9Dn5PIXlhUw1dA1A

**Comment by kcchu** [comment]
I can work around the issue by granting anyuid scc to the `cluster-monitoring-operator` service account. The pods in `openshift-monitoring` seems to be ran happily ever after

```
oc adm policy add-scc-to-user anyuid system:serviceaccount:openshift-monitoring:cluster-monitoring-operator
```

**Comment by vrutkovs** [comment]
This seems to be a monitoring bug - https://bugzilla.redhat.com/show_bug.cgi?id=1904538.

Fixed in 4.6 by https://github.com/openshift/kube-rbac-proxy/pull/37, so it should land in the next stable release
